### PR TITLE
[Installation] Fix MySQL sample data for #__menu

### PR DESCRIPTION
**Pull request by zero-24**
### Summary of Changes

This fixes the lft & rgt values for all our sample data. Thanks @infograf768 

### Testing Instructions

- Install https://github.com/zero-24/joomla-cms/archive/fix_broken_menus.zip as normal Joomla
- Install all sample data
- please try that every entry mainitly frontend behaves as expected.

After the installation please clear the tables and do a complete reinstall

### Expected result

Install should work and all menu entrys should work too

### Actual result

After the lastest #__menu changes we had some issues with the frontend menus 

### Documentation Changes Required

None

### Open Points

As i have no Postgress nor Mssql Envoirment this needs to be done by someone else. I'm going to open a new issue for that and link it from here.

### CODE REVIEW

**Comment by zero-24** [comment]
Issue for postgress and mssql: https://github.com/joomla/joomla-cms/issues/13981

File: installation/sql/mysql/sample_data.sql [hunk 1]
```diff
@@ -91,28 +91,28 @@ INSERT IGNORE INTO `#__menu` (`id`, `menutype`, `title`, `alias`, `note`, `path`
...
+(107, 'usermenu', 'Site Settings', 'site-settings', '', 'site-settings', 'index.php?option=com_config&view=config&controller=config.display.config', 'component', 1, 1, 1, 23, 0, '0000-00-00 00:00:00', 0, 6, '', 0, '{"menu-anchor_title":"","menu-anchor_css":"","menu_image":"","menu_text":1,"page_title":"","show_page_heading":0,"page_heading":"","pageclass_sfx":"","menu-meta_description":"","menu-meta_keywords":"","robots":"","secure":0}', 29, 30, 0, '*', 0);
 
-INSERT IGNORE INTO `#__menu_types` (`id`, `menutype`, `title`, `description`) VALUES
-(1, 'mainmenu', 'Main Menu', 'The main menu for the site'),
-(2, 'usermenu', 'User Menu', 'A Menu for logged-in Users');
+INSERT IGNORE INTO `#___menu_types` (`id`, `asset_id`, `menutype`, `title`, `description`, `client_id`) VALUES
```
Comment (infograf768) [inline installation/sql/mysql/sample_data.sql]: one underscore too much
Should be `#__menu_types`

Comment (infograf768) [inline installation/sql/mysql/sample_data.sql]: Otherwise, looks good

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on be8f148fe113110c16f1c441a57293e60c158bb2<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13979">issues.joomla.org/tracker/joomla-cms/13979</a>.</sub>

**Comment by franz-wohlkoenig** [comment]
> After the installation please clear the tables and do a complete reinstall

@zero-24 Clear all Tables in phpMyAdmin & reinstall of Joomla! 3.7.0-beta?

**Comment by zero-24** [comment]
> @zero-24 Clear all Tables in phpMyAdmin & reinstall of Joomla! 3.7.0-beta?

Yes but the reinstall should be based on my changes so a new install with different sample data of https://github.com/zero-24/joomla-cms/archive/fix_broken_menus.zip

**Comment by franz-wohlkoenig** [comment]
### First install
Test on "English (GB) Sample Data"
#### All Front End Views
404: index.php/single-news-feed
#### Modules
Feed not found: index.php/feed-display
### Second Install
Test on "Blog English (GB) Sample Data"
All works.

**Comment by infograf768** [comment]
@franz-wohlkoenig 
Looks like your fail test is rather using sample_testing.sql and not basic sample_data.sql

We indeed have an error in sample_testing.sql  in the frontendview menutype, as we have no Newsfeed with id4
`(254, 'frontendviews', 'Single News Feed', 'single-news-feed', '', 'single-news-feed', 'index.php?option=com_newsfeeds&view=newsfeed&id=4', 'component', 1, 1, 1, 17, 0, '0000-00-00 00:00:00', 0, 1, '', 0, '{"show_feed_image":"","show_feed_description":"","show_item_description":"","feed_character_count":"0","menu-anchor_title":"","menu-anchor_css":"","menu_image":"","show_page_heading":0,"page_title":"","page_heading":"","pageclass_sfx":"","menu-meta_description":"","menu-meta_keywords":"","robots":"","secure":0}', 165, 166, 0, '*', 0),
`
see
```sql
INSERT IGNORE INTO `#__newsfeeds` (`catid`, `id`, `name`, `alias`, `link`, `published`, `numarticles`, `cache_time`, `checked_out`, `checked_out_time`, `ordering`, `rtl`, `access`, `language`, `params`, `created`, `created_by`, `created_by_alias`, `modified`, `modified_by`, `metakey`, `metadesc`, `metadata`, `xreference`, `publish_up`, `publish_down`, `description`, `version`, `hits`, `images`) VALUES
(17, 1, 'Joomla! Announcements', 'joomla-announcements', 'http://feeds.joomla.org/JoomlaAnnouncements', 1, 5, 3600, 0, '0000-00-00 00:00:00', 1, 1, 1, 'en-GB', '{"show_feed_image":"","show_feed_description":"","show_item_description":"","feed_character_count":"0","newsfeed_layout":"","feed_display_order":""}', '2011-01-01 00:00:01', 42, 'Joomla', '0000-00-00 00:00:00', 0, '', '', '{"robots":"","rights":""}', '', '0000-00-00 00:00:00', '0000-00-00 00:00:00', '', 1, 0, ''),
(17, 2, 'New Joomla! Extensions', 'new-joomla-extensions', 'http://feeds.joomla.org/JoomlaExtensions', 1, 5, 3600, 0, '0000-00-00 00:00:00', 4, 1, 1, 'en-GB', '{"show_feed_image":"","show_feed_description":"","show_item_description":"","feed_character_count":"0","newsfeed_layout":"","feed_display_order":""}', '2011-01-01 00:00:01', 42, 'Joomla', '0000-00-00 00:00:00', 0, '', '', '{"robots":"","rights":""}', '', '0000-00-00 00:00:00', '0000-00-00 00:00:00', '', 1, 0, ''),
(17, 3, 'Joomla! Security News', 'joomla-security-news', 'http://feeds.joomla.org/JoomlaSecurityNews', 1, 5, 3600, 0, '0000-00-00 00:00:00', 2, 1, 1, 'en-GB', '{"show_feed_image":"","show_feed_description":"","show_item_description":"","feed_character_count":"0","newsfeed_layout":"","feed_display_order":""}', '2011-01-01 00:00:01', 42, 'Joomla', '0000-00-00 00:00:00', 0, '', '', '{"robots":"","rights":""}', '', '0000-00-00 00:00:00', '0000-00-00 00:00:00', '', 1, 0, '');
```

For the module we have
`https://community.joomla.org/blogs/community.feed?type=rss as feed display.
This is not a real feed, thus the error.

@zero-24 
The first error should be easy to solve by just changing the id to an existing one.
The second error: I have no idea what is the rss feed for community
We do have it here, normally
https://www.joomla.org/rss-news-feeds.html

but it just does not give a real feed.
It  may be replaced by
`http://feeds.joomla.org/JoomlaMagazine` which does work

**Comment by infograf768** [comment]
@franz-wohlkoenig
We can do a different PR for the issues above.
If the other menu items do work for you with each sample data, then please mark this as tested OK

**Comment by franz-wohlkoenig** [comment]
@infograf768 tested using "English (GB) Sample Data" – isn't this all parts of sampedata? Tested also "Blog English (GB) Sample Data" too.

**Comment by infograf768** [comment]
English (GB) Sample Data does not contain frontend views menutype. ONly English **Testing** sample data does

**Comment by franz-wohlkoenig** [comment]
@infograf768 testing "English Testing sample" without error is enough for "successfull"?

**Comment by infograf768** [comment]
to test, you have to test each sample data proposed.
easy way is to use the same test site instance and then delete the configuration.php and reinstall with a different sample data.

**Comment by franz-wohlkoenig** [comment]
I have tested this item :white_check_mark: successfully on be8f148fe113110c16f1c441a57293e60c158bb2<br /><br />Same 2 Failures like [reported](https://github.com/joomla/joomla-cms/pull/13979#issuecomment-278766838 "")<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13979">issues.joomla.org/tracker/joomla-cms/13979</a>.</sub>

**Comment by infograf768** [comment]
rtc. thanks<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/13979">issues.joomla.org/tracker/joomla-cms/13979</a>.</sub>

**Comment by zero-24** [comment]
@franz-wohlkoenig @infograf768 see: https://github.com/joomla/joomla-cms/pull/14032 for the newsfeeds fixes

**Pull request merged**